hack/ensure-deps/ensure-curl.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script can be used to check your local development environment
+# for necessary dependencies used in TCE
+
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+TCE_CI_BUILD="${TCE_CI_BUILD:-""}"
+
+SUDO_CMD="sudo"
+if [[ "${TCE_CI_BUILD}" == "true" ]]; then
+  SUDO_CMD=""
+fi
+
+if [[ -z "$(command -v zip)" ]]; then
+  if [[ -f "/etc/redhat-release" ]]; then
+    ${SUDO_CMD} yum -y install curl
```

Comment:
Should we use `dnf` now? Does that cut us off from too many RHEL/CentOS versions?

Reply:
we still have some build systems running rhel7 (7.2 specifically - I think that's the oldest)... so we would need to be able to support those. not sure if `dnf` is installed or can be bootstrapped to have it run on that version

Reply:
Ok - probably not worth it to bootstrap dnf, and I think `yum` aliases to it.